code/crates/core-types/src/value.rs

```diff
@@ -54,24 +54,69 @@ impl<Value> NilOrVal<Value> {
     }
 }
 
-/// Defines the requirements for the type of value to decide on.
+/// The `Value` type denotes the value `v` carried by the `Proposal`
```

Comment:
Proposal is a message from the implementation?

If referring to the paper, I would use `PROPOSAL` instead.

Reply:
Yes it refers to the abstract `Proposal` type/trait. Will add a link to the trait.